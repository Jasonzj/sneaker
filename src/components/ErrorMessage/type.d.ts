import { ReactElement } from 'react'
import { DeepMap, FieldError } from 'react-hook-form'

export type Props = {
  error: DeepMap<Record<string, any>, FieldError>
  name: string
  render: (message: string) => ReactElement
}
