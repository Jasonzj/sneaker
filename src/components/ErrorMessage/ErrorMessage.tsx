import React from 'react'
import { Props } from './type'

const ErrorMessage: React.FC<Props> = ({ error, name, render }) => {
  return <>{error[name] && render(error[name].message)}</>
}

export default ErrorMessage
