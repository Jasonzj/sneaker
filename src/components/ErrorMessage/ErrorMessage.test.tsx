import { render, screen } from '@testing-library/react'
import ErrorMessage from './'
import React from 'react'

test('should render correctly with no error', () => {
  const { asFragment } = render(
    <ErrorMessage errors={{}} name='test' render={(message: string) => <p>{message}</p>} />,
  )

  expect(asFragment()).toMatchSnapshot()
})

test('should render correctly with flat error', () => {
  const { asFragment } = render(
    <ErrorMessage
      errors={{ flat: { type: 'flat', message: 'flat' } }}
      name='flat'
      render={(message: string) => <p>{message}</p>}
    />,
  )

  expect(screen.getByText('flat')).toBeInTheDocument()
  expect(asFragment()).toMatchSnapshot()
})

test('should render correctly with flat error and render className', () => {
  const { asFragment } = render(
    <ErrorMessage
      errors={{ flat: { type: 'flat', message: 'flat' } }}
      name='flat'
      render={(message: string) => <p className='flat'>{message}</p>}
    />,
  )

  expect(screen.getByText('flat')).toHaveClass('flat')
  expect(asFragment()).toMatchSnapshot()
})
