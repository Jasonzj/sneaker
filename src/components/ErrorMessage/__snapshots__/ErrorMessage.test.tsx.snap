// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly with flat error 1`] = `
<DocumentFragment>
  <p>
    flat
  </p>
</DocumentFragment>
`;

exports[`should render correctly with flat error and render className 1`] = `
<DocumentFragment>
  <p
    class="flat"
  >
    flat
  </p>
</DocumentFragment>
`;

exports[`should render correctly with no error 1`] = `<DocumentFragment />`;
